mobile/lib/features/trade/domain/position.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:get_10101/features/trade/domain/contract_symbol.dart';
+import 'package:get_10101/features/trade/domain/direction.dart';
+import 'package:get_10101/features/trade/domain/leverage.dart';
+import 'package:get_10101/common/domain/model.dart';
+import 'package:get_10101/bridge_generated/bridge_definitions.dart' as bridge;
+
+enum PositionState {
+  open,
+
+  /// once the user pressed button to close position the button should be disabled otherwise the user can click it multiple times which would result in multiple orders and an open position in the other direction
+  closing;
```

Comment:
May be worth copying the comment in the rust model to explain why there is no "closed" variant. Also may consider a different name as "PositionState" seems to imply the state of a position,which seems to include closed ones, as opposed really just representing manipulable positions / non-closed positions :man_shrugging:  Maybe I am still just thinking in the itchysats way, though

Reply:
For me this specific comment contains an implementation detail for the UI - we are talking about the behavior of a button here. 

There is rustdoc in the rust model (in `api_model` - we could move the comments to the internal model because that represents the state machine in more detail).
I will see to harmonize that in a follow up, i.e. always describe the state machine in the feature model and reference it from the api model.

```rust
pub enum PositionState {
    /// The position is open
    ///
    /// Open in the sense, that there is an active position that is being rolled-over.
    /// Note that a "closed" position does not exist, but is just removed.
    /// During the process of getting closed (after creating the counter-order that will wipe out
    /// the position), the position is in state "Closing".
    ///
    /// Transitions:
    /// Open->Closing
    Open,
    /// The position is in the process of being closed
    ///
    /// The user has created an order that will wipe out the position.
    /// Once this order has been filled the "closed" the position is not shown in the user
    /// interface, so we don't have a "closed" state because no position data will be provided to
    /// the user interface.
    Closing,
}
```